lib/utils/config.ts

```diff
@@ -17,16 +17,17 @@ import Promise = require('bluebird');
 import ResinSdk = require('resin-sdk');
 import _ = require('lodash');
 import deviceConfig = require('resin-device-config');
+import * as semver from 'resin-semver';
 
 const resin = ResinSdk.fromSharedOptions();
 
 export function generateBaseConfig(
 	application: ResinSdk.Application,
-	options: { appUpdatePollInterval?: number },
+	options: { version?: string; appUpdatePollInterval?: number },
 ) {
 	options = _.mapValues(options, function(value, key) {
 		if (key === 'appUpdatePollInterval') {
-			return value! * 60 * 1000;
+			return options[key]! * 60 * 1000;
```

Comment:
I actually think this could be simplified more with:
```typescript
if (options.appUpdatePollInterval) {
	options.appUpdatePollInterval = options.appUpdatePollInterval * 60 * 1000
}
```
and just drop the `mapValues` altogether (could use `_.clone` if the cloning was a desired side-effect)

Reply:
I do indeed want the clone, rather than mutating my arguments. I think the latter is a nice way of combining them both, I've gone with that (but also, I feel like we're bike shedding a bit here, let's just ship it).